model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -547,11 +540,17 @@ public ClientModel getCreatedClient() {
     }
 
     @Override
-    public List<ClientModel> getClients(RealmModel realm) {
+    public List<ClientModel> getClients(RealmModel realm, Integer firstResult, Integer maxResults) {
         TypedQuery<String> query = em.createNamedQuery("getClientIdsByRealm", String.class);
+        if (firstResult != null && firstResult > 0) {
+            query.setFirstResult(firstResult);
+        }
+        if (maxResults != null && maxResults > 0) {
+            query.setMaxResults(maxResults);
+        }
         query.setParameter("realm", realm.getId());
         List<String> clients = query.getResultList();
-        if (clients.isEmpty()) return Collections.EMPTY_LIST;
+        if (clients.isEmpty()) return Collections.emptyList();
```

Comment:
Please don't make unnecessary changes (Collections.EMPTY_LIST is perfectly valid)

Reply:
Sure, I changed it back. I'm using an IDE for the development and it was giving me an `Unchecked assignment` warning, since `Collections.EMPTY_LIST` returns the non-generic version.